megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -618,6 +621,99 @@ protected void calculateFiringTurn() {
                     logger.debug(shooter.getDisplayName() + " - Detailed Best Firing Plan: " +
                             plan.getDebugDescription(true));
 
+                    // If the shooter is a ground unit (including VTOLs and infantry) and only
+                    // shooting at a single Mech target with all weapons, consider making an
+                    // aimed shot if the target is shut down or the attacker has a targeting
+                    // computer. Alternatively, consider using the called shots optional rule to
+                    // adjust the hit table to something more favorable.
+
+                    boolean isShutdownShot = false;
```

Comment:
Let's take this newly added logic and put it into a separate function to avoid bloating getFiringTurn()

Reply:
I considered it but not entirely certain how to factor it out, given it has to return multiple settings - whether to test for an aimed shot and/or called shots, get an aimed shot location, get a called shot direction, and setting an appropriate maximum to-hit number.  Thoughts?